sklearn/naive_bayes.py

```diff
@@ -420,42 +438,54 @@ def _partial_fit(self, X, y, classes=None, _refit=False, sample_weight=None):
 
         first_call = _check_partial_fit_first_call(self, classes)
         X, y = validate_data(self, X, y, reset=first_call)
+        xp, _, device_ = get_namespace_and_device(X)
         if sample_weight is not None:
+            sample_weight = xp.asarray(sample_weight, device=device_)
             sample_weight = _check_sample_weight(sample_weight, X)
 
+        xp_y, _ = get_namespace(y)
         # If the ratio of data variance between dimensions is too small, it
         # will cause numerical errors. To address this, we artificially
         # boost the variance by epsilon, a small fraction of the standard
         # deviation of the largest dimension.
-        self.epsilon_ = self.var_smoothing * np.var(X, axis=0).max()
+        self.epsilon_ = self.var_smoothing * xp.max(xp.var(X, axis=0))
 
         if first_call:
             # This is the first call to partial_fit:
             # initialize various cumulative counters
             n_features = X.shape[1]
-            n_classes = len(self.classes_)
-            self.theta_ = np.zeros((n_classes, n_features))
-            self.var_ = np.zeros((n_classes, n_features))
+            n_classes = self.classes_.shape[0]
+            max_float_dtype = _max_precision_float_dtype(xp, device=device_)
```

Comment:
I think we should use the dtype of `X` instead of attempting to allocating `float64` precision arrays in case where all the computation happens at `float32` in `_update_mean_variance`.

This change should be documented in the changelog and tests of for the dtypes of the fitted attributed should be updated/added.

Reply:
Indeed, let's use something like `dtype = _find_matching_floating_dtype(X, xp)` then.